### ISSUE TITLE
Allow for increased customizability for VideoStreamOptions

### ISSUE THREAD

**Issue by MarcusTL12** [MakieOrg/Makie.jl#2938]
Currently all the options for video are quite restrictive. Personally I would like to use hardware acceleration for example. Would it be possible/easy/wanted to have an option to specify raw ffmpeg flags to the VideoStreamOptions (and record function)? This way it would be possible to tweak the encoding options much more control.

**Comment by asinghvi17** [comment]
Sure, I don't see that being too much of an issue.  Which specific flags would you like to use?  I was under the impression that FFMPEG already uses hardware acceleration where it can...

**Comment by MarcusTL12** [comment]
Yes, FFMPEG can definitely use hardware acceleration, but it has to be specified as the codec to be used (e.g. -c:v h264_nvenc for nvidia gpu h264 encoding). Maybe a codec option would be nice to have. The extra_options field would also allow users to test various obscure FFMPEG flags without it having to be specified in the Makie.jl source code.
